projects/scion/microfrontend-platform/src/lib/platform.model.ts

```diff
@@ -139,6 +159,20 @@ export interface Capability {
    *   Is allowed as qualifier property value only and matches intents regardless of having or not having that property.
    */
   qualifier?: Qualifier;
+  /**
+   * The params object is a dictionary of arbitrary key-value pairs to further describe the capability.
+   * By defining these additional parameters, it becomes much more obvious for a micro application how to successfully invoke this capability.
+   *
+   * In contrast to the qualifier, params have no effect on intent routing.
+   *
+   * The params object allows using wildcards (such as `*` or `?`) to match multiple intents simultaneously.
+   *
+   * - **Asterisk wildcard character (`*`):**\
+   *   Is allowed as params property value only and requires intents to contain that property, but with any value allowed (except for `null` or `undefined` values).
+   * - **Optional wildcard character (?):**\
+   *   Is allowed as params property value only and matches intents regardless of having or not having that property.
+   */
+  params?: Params;
```

Comment:
Specifying the param's datatype would allow us to validate it in the message broker (only if specified). Of course, it would only be a `typeof` check.

Reply:
It's true that params have a different semantic than the qualifier. But from a user perspective (a user that defines or consumes a capability), both the qualifier and the params are part of the contract. The fact that params don't have any effect on intent routing is secondary, because missing or superfluous parameters will still result in the intent being rejected. In other words, the params of the intent have to match the params of the capability, the same way the qualifiers have to match.

- So why use a different syntax for the params?
- Why should a parameter have a description, but not the qualifier (or qualifier key)? It might not always be obvious what a qualifier does or what data type it needs.
- Do we really want to allow complex objects as parameters? Won't it make the contract much more difficult to understand (and less precise and thus more error-prone)? E.g. the capability provider is expecting a `User` (of type `object`) and the consumer is sending a `Person`. The intent won't be rejected, but the request might still fail. Shouldn't we enforce sending complex objects in the message body (or header) and leave validation to the user?

Reply:
It is absolutely okay for me to have a different notation for qualifiers and parameters because they serve a different purpose: qualifying a capability vs. passing data. And yes, you are right. Parameters are not intended to pass complex data along with the intent.

Proposal: We concentrate on the essential: I.e. two separate arrays for parameter names, one for required and one for optional parameters. We do not yet provide an API to describe parameters or define their datatype.